### PR TITLE
feat added Auth

### PR DESCRIPTION
**Pull request by Himanshu-Dhawale**
1. Added Login feature
2. Added Signup feature

### CODE REVIEW

**Review by ayush65 — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -1,43 +1,20 @@
...
-            rel="noreferrer"
-          >
-            Contribute
-          </a>
-        </div>
-      </header>
-    </div>
-  );
+ <Routes>       
+{/* <Route path="/" element={<Home />} /> */}
```
Comment (Sharmaryan) [inline src/App.js]: remove this code

File: src/Login/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import {
+  useState
+} from "react";
```
Comment (Sharmaryan) [inline src/Login/Login.jsx]: you can write it this way ```import React, {
  useState
} from "react"; ```

File: src/Login/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  });
+  const { user,setUser } = useAuth();
+
+    const changeHandler = (e) => {
+      setUserDetail((previousState) => ({
+        ...previousState,
+        [e.target.name]: e.target.value,
+      }));
+    };
+console.log(user)
```
Comment (Sharmaryan) [inline src/Login/Login.jsx]: remove console log

File: src/Login/login.css [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  
+  .login {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    margin: auto;
+    padding: 2rem 3rem;
+    box-shadow: #8838ea 0px 0px 7px;
+    border-radius: 15px;
```
Comment (Sharmaryan) [inline src/Login/login.css:19]: I think there is no need to write px after 0

File: src/Login/login.css [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  
+  .login__remember {
+    margin: 0.8rem;
+    cursor: pointer;
+  }
+  
+  .login__button {
+    padding: 0.5rem 0;
+    margin: 10% 0 0 0;
+    border-radius: 5px;
```
Comment (Sharmaryan) [inline src/Login/login.css:59]: be consistent, either give rem here or px instead of %

File: src/Signup/Signup.jsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  const {user ,setUser } = useAuth();
+  const signUpChangeHandler = (e) => {
+    setSignUpUser((previousState) => ({
+      ...previousState,
+      [e.target.name]: e.target.value,
+    }));
+  };
+  const signUpClickHandler = async (e) => {
+   e.preventDefault()
+    console.log("hello");
```
Comment (Sharmaryan) [inline src/Signup/Signup.jsx]: remove this console log

File: src/contexts/auth-context.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { createContext, useContext, useState } from "react";
+
+const CreateAuth = createContext({ user: null, token: null });
+
```
Comment (Sharmaryan) [inline src/contexts/auth-context.js:4]: instead of CreateAuth, a better name could be AuthContext

**Review by Sharmaryan — APPROVED** [review]
left few suggestions

**Pull request merged**